dummyfs/dir.c

```diff
@@ -35,6 +35,8 @@ int dir_find(dummyfs_object_t *dir, const char *name, oid_t *res)
 	if (e == NULL)
 		return -ENOENT;
 
+	dirname = strdup(name);
```

Comment:
Still no check for ENOMEM

Reply:
Yeah, I think dummyfs needs some careful review. I believe the current approach (always trying to `realloc` chunks) may result in performance slowdowns (hidden `memcpy` in `realloc` implementation).

Also - we need at least some basic functionality checks for file systems (this error was triggered every time a file was truncated to 0 - so for example deleted). Unfortunately we can't test `dummyfs` on `ia32-generic` due to  phoenix-rtos/phoenix-rtos-project#114)

Reply:
I agree. I think that dummyfs makes writes at NULL pointer - we found traces of it only when dummyfs has been started...